### ISSUE TITLE
[Bug]: INSTANA_EXTRA_HTTP_HEADERS does nothing

### ISSUE THREAD

**Issue by martinei** [instana/go-sensor#338]
### Problem Description

Observed: 
I set INSTANA_EXTRA_HTTP_HEADERS for my go-service to some value  to collect additional headers, but it had no effect. Looking at
https://github.com/instana/go-sensor/blob/master/agent.go#L367 it believe it is override by whatever the Agent send.

Expected:
Headers set in INSTANA_EXTRA_HTTP_HEADERS are collected.

### Go Version

1.16

### go.mod

```shell
I don't think this is necessary
```

### go env

```shell
I don't think this is necessary
```

**Comment by sanojsubran** [comment]
Hello @martinei,
 
Thanks for reporting the issue to us. We analysed the reported issue and we identified one issue regarding the configuration of custom headers and the fix has been added to the latest release[[1.43.5](https://github.com/instana/go-sensor/releases/tag/v1.43.5)]. Currently there is a priority in which the source of the custom headers will be determined and it is as follows: agent config < via sdk < env variable. So kindly choose the source appropriately which will overwrite the ones with a lesser priority.

Kindly check with our latest release.
 
Kind regards,
Sanoj Subran
